src/pudl/transform/ferc1.py

```diff
@@ -2931,6 +2926,82 @@ def merge_xbrl_metadata(
+            "other_accounts_details",  # Unstructured
+            "accumulated_depreciation_distribution",
+            "accumulated_depreciation_general",
+            "accumulated_depreciation_hydraulic_production_conventional",
+            "accumulated_depreciation_hydraulic_production_pumped_storage",
+            "accumulated_depreciation_nuclear_production",
+            "accumulated_depreciation_other_production",
+            "accumulated_depreciation_regional_transmission_and_market_operation",
+            "accumulated_depreciation_steam_production",
+            "accumulated_depreciation_transmission",
+            "ending_balance_functional",
+        ]
+        df = super().process_dbf(df).assign(utility_type="electric")
+        df = df.loc[~df.ferc_account_label.isin(excluded_labels)]
+        return df
+
+    @cache_df("process_instant_xbrl")
+    def process_instant_xbrl(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Pre-processing required to make the instant and duration tables compatible.
+
+        Each year the plant account balances are reported twice, in two separate
+        records: one for the end of the previous year, and one for the end of the
+        current year, with appropriate dates for the two year ends. Here we are
+        reshaping the table so that we instead have two columns: ``starting_balance``
+        and ``ending_balance`` that both pertain to the current year, so that all of
+        the records pertaining to a single ``report_year`` can be identified without
+        dealing with the instant / duration distinction.
+        """
+        df = super().process_instant_xbrl(df)
+        df["year"] = pd.to_datetime(df["date"]).dt.year
+        df.loc[df.report_year == (df.year + 1), "balance_type"] = "starting_balance"
+        df.loc[df.report_year == df.year, "balance_type"] = "ending_balance"
+        if not df.balance_type.notna().all():
+            raise ValueError(
+                f"Unexpected years found in the {self.table_id.value} table: "
+                f"{df.loc[df.balance_type.isna(), 'year'].unique()}"
+            )
+        df = (
+            df.drop(["year", "date"], axis="columns")
+            .set_index(
+                [
+                    "entity_id",
+                    "report_year",
+                    "electric_plant_classification_axis",
+                    "utility_type_axis",
+                    "balance_type",
+                ]
+            )
+            .unstack("balance_type")
+        )
+        df.columns = ["_".join(items) for items in df.columns.to_flat_index()]
+        df = df.reset_index().rename(
+            columns={
+                "accumulated_provision_for_depreciation_of_electric_utility_plant_ending_balance": "ending_balance_accounts_electric_plant",
+                "accumulated_provision_for_depreciation_of_electric_utility_plant_starting_balance": "starting_balance_electric_plant",
+            }
+        )
```

Comment:
Not a fan of this embedded rename, but because of the order in which the `wide_to_tidy` and `rename_columns` calls happen, I couldn't do it in either the `xbrl` or `instant_xbrl` rename steps.  Is there a better way?

Reply:
Since you are currently not doing an instant_xbrl rename, you could swap the order of `process_instant_xbrl`, so it would be:
* unstack start/end balance
* super

until you merge dev in since austin made this unstack a general transformer, you'll fully override `process_instant_xbrl` with:
* unstack
* rename

Reply:
also is `ending_balance_accounts_electric_plant` supposed to be `ending_balance_electric_plant`?

Reply:
So you're suggesting redefining the whole `process_instant_xbrl` method, to do the same operations but in a different order?

Reply:
Yes -- and use the `rename_columns` with instant_xbrl as the stage, but in a different order.

Reply:
Implemented, fully overriding the`process_instant_xbrl()` method and using the `instant_xbrl` rename stage.